CMake/cdat_modules/openssl_external.cmake

```diff
@@ -0,0 +1,37 @@
+set (OPENSSL_SOURCE_DIR "${CMAKE_CURRENT_BINARY_DIR}/build/openssl")
+set (OPENSSL_INSTALL_DIR "${cdat_EXTERNALS}")
+
+execute_process (COMMAND uname -s COMMAND tr -d '\n'
+                 OUTPUT_VARIABLE HOST)
+STRING (TOLOWER ${HOST} HOST)
+execute_process (COMMAND uname -m COMMAND tr -d '\n'
+                 OUTPUT_VARIABLE ARCHITECTURE)
+
+get_filename_component (COMPILER "${CMAKE_C_COMPILER}" NAME_WE)
+
+if (APPLE)
+  if (ARCHITECTURE MATCHES "64$")
+    set (HOST "${HOST}64")
+  endif ()
+  set (COMPILER "cc")
```

Comment:
Those are options required by the OpenSSL configure script. Documentation [here](https://wiki.openssl.org/index.php/Compilation_and_Installation)